crates/bevy_ecs/src/system/query.rs

```diff
@@ -543,6 +543,17 @@ where
             >())),
         }
     }
+
+    /// Returns true if this query contains no elements.
+    pub fn is_empty(&self) -> bool {
+        // SAFE: system runs without conflicts with other systems.
+        // same-system queries have runtime borrow checks when they conflict
+        let iter = unsafe {
+            self.state
+                .iter_unchecked_manual(self.world, self.last_change_tick, self.change_tick)
+        };
+        iter.peekable().peek().is_none()
```

Comment:
why go through a `peekable` instead of `next().is_none()`?

Reply:
Because then you have to make `iter` `mut` 😨 

All kidding aside, `.next().is_none()` is probably better :)